### PR TITLE
Migrating changes from our branch to openPDF Part I

### PR DESCRIPTION
**Pull request by Lonzak**
We do maintain our own branch of iText.
All these changes should be migrated to openPDF.
This is a first (tiny) step in doing so

### CODE REVIEW

**Comment by Lonzak** [comment]
No breaking changes AFAIK. The changes fixes Bugs and offers a broader usage. 

E.g.
Adding convenience or missing methods:
* public boolean setFieldProperty(Item item, String name, int value, int inst[]) in AcroFields
* public boolean isAppend() in AcroFields
* added methods to add the modification date (in PdfDocument and PdfAppearance)
* added mappingName and alternateFieldName methods (BaseField.java)
* getFontDescriptor() (DocumentFont)
* addFlags(int flags) in PdfAnnotation
* createAppearance(...) in PdfAppearance - the possibility to create a new appearance for an existing form field
* remove(PdfObject object) in PdfArray - ability to remove an objects from an array
* new Constructor for PdfDate: public PdfDate(String d)
* clear() method for PdfDictionary()
* createTextField() (PdfFormField)
* getPageSizeWithRotation() in PdfReader
* added SignatureBuildProperty in PdfSignature
* SignatureRenderGraphic to PdfSignatureAppearance
* several new methods in RadioCheckField
* ...

Adding missing PDF Keys, massively documented the keys / **JavaDoc**, Added new algorithms
* PdfName class
* PdfPKCS7
* Webcolors, AcroFields, 

Refactored and corrected:
* Constants in Element.java

**Comment by RainerGanss** [comment]
ehm, this PR looks awesome! Why is no one merging this?

File: openpdf/src/main/java/com/lowagie/text/Document.java [hunk 1]
```diff
@@ -625,6 +626,47 @@ public boolean addCreationDate() {
...
+     *
+     * @return  <CODE>true</CODE> if successful, <CODE>false</CODE> otherwise
+     */
+    public boolean addModificationDate() {
+        try {
+            return add(new Meta(Element.MODIFICATIONDATE, new PdfDate().toString()));
+        } catch (DocumentException de) {
+            throw new ExceptionConverter(de);
+        }
+    }
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Document.java:655]: This is not a must have method, can be replaced by addModificationDate(PdfDate) and passing in current date.

API should be as simpler/cleaner as possible

File: openpdf/src/main/java/com/lowagie/text/Element.java [hunk 2]
```diff
@@ -79,111 +79,114 @@ public interface Element {
     // static membervariables (meta information)
 
     /** This is a possible type of <CODE>Element</CODE>. */
-    int HEADER = 0;
+    public static final int HEADER = 0;
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Element.java:83]: Interface fields is public by default, final static helps though

File: openpdf/src/main/java/com/lowagie/text/Element.java [hunk 3]
```diff
@@ -312,7 +315,7 @@ public interface Element {
      * @return a type
      */
 
-    int type();
+    public int type();
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Element.java:319]: Java interface is public by default, NO need to declare it as public, which makes code heavy.

File: openpdf/src/main/java/com/lowagie/text/Element.java [hunk 4]
```diff
@@ -321,29 +324,30 @@ public interface Element {
      * @return    true if this is a 'content' element; false if this is a 'metadata' element
      */
 
-    boolean isContent();
+    public boolean isContent();
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Element.java:328]: Java interface is public by default, NO need to declare it as public, which makes code heavy.

File: openpdf/src/main/java/com/lowagie/text/Element.java [hunk 5]
```diff
@@ -321,29 +324,30 @@ public interface Element {
...
     
     /**
      * Checks if this element is nestable.
      * @since    iText 2.0.8
      * @return    true if this element can be nested inside other elements.
      */
 
-    boolean isNestable();
+    public boolean isNestable();
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Element.java:336]: Java interface is public by default, NO need to declare it as public, which makes code heavy.

File: openpdf/src/main/java/com/lowagie/text/Element.java [hunk 6]
```diff
@@ -321,29 +324,30 @@ public interface Element {
...
     
     /**
      * Gets all the chunks in this element.
      * 
      * @return an <CODE>ArrayList</CODE>
      */
 
-    ArrayList<Element> getChunks();
+    public ArrayList<Element> getChunks();
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Element.java:344]: Java interface is public by default, NO need to declare it as public, which makes code heavy.

File: openpdf/src/main/java/com/lowagie/text/Element.java [hunk 7]
```diff
@@ -321,29 +324,30 @@ public interface Element {
...
 
     /**
      * Gets the content of the text element.
      * 
      * @return a type
      */
 
-    String toString();
+    @Override
+    public String toString();
```
Comment (wayne-zhang) [inline openpdf/src/main/java/com/lowagie/text/Element.java:352]: Java interface is public by default, NO need to declare it as public, which makes code heavy.

File: openpdf/src/main/java/com/lowagie/text/Element.java (hunk 4 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/Element.java:328]: sure

**Comment by Lonzak** [comment]
Thanks for the merge!

**Comment by andreasrosdal** [comment]
@Lonzak  You are welcome. More improvements to OpenPDF from you is welcome.

**Pull request merged**